pipeline/runtime/step_executor.go

```diff
@@ -396,24 +396,58 @@ func setPrevStepExportEnvs(r *api.StartStepRequest) {
 }
 
 func (e *StepExecutor) sendStepStatus(r *api.StartStepRequest, response *api.VMTaskExecutionResponse) {
-	jsonData, err := json.Marshal(response)
-	if err != nil {
-		logrus.WithField("id", r.ID).Errorln("Error marshaling struct:", err)
```

Comment:
Let's keep this log line

Reply:
This is still printed here:
`logrus.WithField("id", r.ID).WithError(err).Errorln("failed to send step status")`

new `sendStatus` method (which wraps this marshaling) is now returning err which then gets printed in the above line. If I kept the  `logrus.WithField("id", r.ID).Errorln("Error marshaling struct:", err) `, it would print the error message twice

Reply:
Marshalling errors are getting swallowed by `sendStatus` function